lib/do_image/do_tag.js

```diff
@@ -0,0 +1,130 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright 2020 Joyent, Inc.
+ *
+ * `triton image tag ...`
+ */
+
+var vasync = require('vasync');
+
+var common = require('../common');
+var errors = require('../errors');
+var mat = require('../metadataandtags');
+
+// ---- the command
+
+function do_tag(subcmd, opts, args, cb) {
+    if (opts.help) {
+        this.do_help('help', {}, [subcmd], cb);
+        return;
+    } else if (args.length < 1) {
+        cb(new errors.UsageError('incorrect number of args'));
+        return;
+    }
+
+    var log = this.top.log;
+    var tritonapi = this.top.tritonapi;
+
+    vasync.pipeline({arg: {cli: this.top}, funcs: [
+        common.cliSetupTritonApi,
+
+        function gatherTags(ctx, next) {
+            mat.tagsFromSetArgs(opts, args.slice(1), log, function (err, tags) {
+                if (err) {
+                    next(err);
+                    return;
+                }
+                log.trace({tags: tags || '<none>'},
+                    'tags loaded from opts and args');
+                ctx.tags = tags;
+                next();
+            });
+        },
+        function updateImageTags(ctx, next) {
+            log.trace({dryRun: opts.dry_run, tags: ctx.tags},
+                'image tag');
+
+            if (opts.dry_run) {
+                next();
+                return;
+            }
+
+            tritonapi.updateImage({
+                image: args[0],
+                fields: {
+                    tags: ctx.tags
```

Comment:
FYI - this gives me a traceback when I don't define any tags:
```
$ ./bin/triton image tag 506cdbc2

/Users/toddw/joyent/node-triton/node_modules/assert-plus/assert.js:22
    throw new assert.AssertionError({
    ^

AssertionError [ERR_ASSERTION]: opts.fields tags (object) is required
    at new AssertionError (internal/assert.js:268:11)
    at _toss (/Users/toddw/joyent/node-triton/node_modules/assert-plus/assert.js:22:11)
    at Function.out.(anonymous function) [as object] (/Users/toddw/joyent/node-triton/node_modules/assert-plus/assert.js:117:17)
    at /Users/toddw/joyent/node-triton/lib/cloudapi2.js:1157:25
    at Array.forEach (<anonymous>)
    at CloudApi.updateImage (/Users/toddw/joyent/node-triton/lib/cloudapi2.js:1154:30)
    at Object.cloudApiUpdateImage [as func] (/Users/toddw/joyent/node-triton/lib/tritonapi.js:1110:27)
    at Immediate.<anonymous> (/Users/toddw/joyent/node-triton/node_modules/vasync/lib/vasync.js:213:20)
    at runCallback (timers.js:696:18)
    at tryOnImmediate (timers.js:667:5)
```

Reply:
Good call. I've added an earlier error message notifying regarding the need to provide at least a single key=value pair